src/components/interactive-map.js

```diff
@@ -94,12 +99,8 @@ const propTypes = Object.assign({}, StaticMap.propTypes, {
   getCursor: PropTypes.func,
 
   // A map control instance to replace the default map controller
-  // The object must expose one property: `events` as an array of subscribed
-  // event names; and two methods: `setState(state)` and `handle(event)`
-  controller: PropTypes.shape({
-    events: PropTypes.arrayOf(PropTypes.string),
-    handleEvent: PropTypes.func
-  })
+  // The object must expose a method: `setOptions(opts)`
+  controller: PropTypes.instanceOf(MapController)
```

Comment:
Follow-up: should add a babel transform to convert flow types to proptypes for the dist file? That way we get type safety by default but don't break API

Reply:
FWIW, last time I looked into auto-generating proptypes from flowtypes it had a lot of rough edges, see this: https://github.com/uber-web/baseui/issues/47

Maybe it has improved since then. Safest bet is just to manually define propTypes, and surface area of react-map-gl's public API is relatively small.